boltdesignsystem/bolt
# Add new `renderWC` helper function to Jest to simplify tests checking for WC rendered HTML

## Jira
http://vjira2:8080/browse/BDS-2119

Note: this PR is required by #1842 so our Jest tests don't run into any hickups.

## Summary
This PR adds a new function to our Jest testing helpers that significantly simplifies getting raw web component markup rendered in Puppeteer for testing.

This PR _also_ helps us prep for the upstream web performance updates (ex. getting `<ssr-keep>` support added globally) which would have required us to make similar component test updates.

## Details
For example, instead of having to write all this:

```
const innerHTML = await page.evaluate(async results => {
      const wrapper = document.createElement('div');
      wrapper.innerHTML = results.html;
      document.body.appendChild(wrapper);
      const button = document.querySelector('bolt-button');
      await button.firstUpdated;
      return button.renderRoot.innerHTML;
    }, results);
```

Now all we have to do is write this:
```
const { innerHTML } = await renderWC('bolt-button', results.html, page);

// for reference, this is basically the full set of params + data that can get returned
// const { innerHTML, outerHTML } = await renderWC(tagName, htmlToRender, puppeteerPageVariable);
```

## How to test
- Check out the new testing helper
- Review changes to existing Jest tests
- Confirm Jests tests all passing as expected
